src/libponyc/pass/docgen.c

```diff
@@ -403,6 +407,30 @@ static void doc_type_list(docgen_t* docgen, ast_t* list, const char* preamble,
   fprintf(docgen->type_file, "%s", postamble);
 }
 
+static bool is_for_testing(const char* name, ast_t* list)
+{
+  assert(name != NULL);
+  assert(list != NULL);
+
+  if (strncmp(name, "_Test", 5) == 0) return true;
+
+  if(ast_id(list) == TK_NONE)
+    return false;
+
+  for(ast_t* p = ast_child(list); p != NULL; p = ast_sibling(p))
+  {
+    if (ast_id(p) == TK_NOMINAL)
+    {
+      AST_GET_CHILDREN(p, package, id, tparams, cap, ephemeral);
```

Comment:
what `AST_GET_CHILDREN` does is:

1. Operates on a single ast node (the first argument).
2. Declares new variables that have the same name as all the following arguments.
3. Gets the children of the first argument, in order, and assigns them to these new variables, in order.